### ISSUE TITLE
TextBox growing - eternal loop

### ISSUE THREAD

**Issue by hanskarlsen** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#869]
Hi - sorry for not being able to post a sample just yet. Maybe you see the issue straight away.

The issue is in The TextBox - in the Control Template. I have fixed the issue in an override by removing this:

 <Line x:Name="DashedLine" VerticalAlignment="Bottom" Visibility="Hidden"
                                  StrokeThickness="1.25" StrokeDashArray="1,2.5" StrokeDashCap="Round"
                                  X1="0" X2="{Binding ActualWidth, ElementName=border}" Y1="0" Y2="0" 
                                  Stroke="{TemplateBinding BorderBrush}" Opacity="0.56" />

The issue shows up when you have TextBoxes in a Grid in certain configurations (have not been able to pinpoint exact repro) - the TextBox has HorizontalAlign Stretch.

What happens (when it happens) is this i think:
The DashedLine  X2 is updated from the border.ActualWidth - and in doing so somehow triggers the stretched textbox to update width -> border.ActualWidth follows -> DashedLine X2 follows -> round-we-go....

On screen it looks like the UI grid grows a bit second by second. 

/Hans

**Comment by Keboo** [comment]
@hanskarlsen was there more information (images?) that were suppose to be attached to this? I am not sure how to reproduce the issue.

**Comment by hanskarlsen** [comment]
Hi - I have been trying to reproduce in a test project but have not been able to yet. Seems there is something I do not understand fully yet.

`                            <Line x:Name="DashedLine" VerticalAlignment="Bottom" Visibility="Hidden"
                                  StrokeThickness="1.25" StrokeDashArray="1,2.5" StrokeDashCap="Round"
                                  X1="0" X2="{Binding ActualWidth, ElementName=border}" Y1="0" Y2="0" 
                                  Stroke="{TemplateBinding BorderBrush}" Opacity="0.56" />
`

It is the above xaml that solves the issue when removed.
In some circumstances WPF seems to get into recursion - the line is updated - it somehow effects the ActualWidth - the line is updated - affects ActualWidth - for eternity. I can provide a video of the effect if you think it might help?

**Comment by hanskarlsen** [comment]
![2018-01-04_12h05_50](https://user-images.githubusercontent.com/2099327/34561560-f1e5088c-f14a-11e7-8ab1-d5b8d10fadd2.gif)
![2018-01-04_12h08_36](https://user-images.githubusercontent.com/2099327/34561565-f50cb050-f14a-11e7-964d-ae06f0beb745.gif)

First video has the TextBox style intact - second has the Line commented out

**Comment by Keboo** [comment]
Interesting. Is it possible to also share your TextBox declarations and the XAML that contains them? I find it interesting that only the text boxes on the left side of the screen grow, while the ones on the right do not. Are there any Bindings to a Width property on that page?

**Comment by hanskarlsen** [comment]
I agree that it is strange that it was only the left side that did this. While trying to understand it I moved around and removed stuff - then it stopped - but under certain (cannot pinpoint which) conditions it starts.
These screens are part of a larger system - most screens do not behave like this - but we have found 2 that does. These 2 look almost the same.

The screens are created by code - by following a declaration (MDriven - ViewModel). The code itself does not do anything strange - but it makes it hard to extract the xaml to a repeating test case. I see if I can do it somehow...
